tests/devices/test_default_qubit.py

```diff
@@ -1989,3 +1661,136 @@ def test_s_inverse():
 
         expected = np.array([1., -1.j]) / np.sqrt(2)
         assert np.allclose(dev.state, expected, atol=tol, rtol=0)
+
+@pytest.mark.parametrize("inverse", [True, False])
+class TestApplyOperationUnit:
+    """Unit tests for the internal _apply_operation method."""
+
+    def test_internal_apply_ops_case(self, inverse):
+        """Tests that if we provide an operation that has an internal
+        implementation, then we use that specific implementation."""
```

Comment:
Nice! I would find the test clearer if it would use a name like `SupportedGate` instead of `PauliX`...it took me a while to understand that you redefine that gate!

Reply:
Oh, good point! changed to have `supported_gate_application` as the new method and also added monkeypatching to make it a cleaner solution. (Note that the gate itself is not redefined, rather the internal function that `default.qubit` uses to apply `PauliX` is, adding this to the docstring.)